DESIGN.md

```diff
@@ -0,0 +1,108 @@
+# Vanagon
+- Note that this is an incomplete doc, documentation on all aspects of vanagon to come
+
+## Vanagon for the Windows platform
+
+### Overview
+Vanagon was adapted for the windows platform using WiX. At a very high level each component of the project gets built on a build system, then gets moved to a staging directory where we use the WiX toolset to build an MSI. This workflow is consistent with the rest of vanagon, however the WiX toolset is not so simple that there are no caveats, specifically:
+
+* Vanagon does not contain functionality to handle registry searching
+* Vanagon does not contain functionality to handle the MSI UI
+* If the project requires MSI custom actions (such as those to search the registry, set properties used in the MSI etc.) Vanagon has no functionality to handle those
+* For directories and files that require special attention (Such as a requirement for an app data directory to persist after uninstall) vanagon cannot handle them
+* Vanagon cannot handle the service files required by WiX for an MSI
+* Vanagon cannot handle the localization engine that MSIs use
+
+We deal with these special cases by allowing the user to provide their own WiX files inside the vanagon project. This way the user can use the WiX toolset to solve problems like registry searching and service files.
+
+It’s possible many of these edge cases can be remedied in future releases of vanagon, such as the special casing of directories and files. But the benefit of “consistency” may not be enough to drive change for other cases, specifically because for things like custom actions the changes to vanagon would be so drastic and edge cased for windows, vanagon would start to look inconsistent with itself.
+
+### Vanagon classes, functions, and WiX
+We made no major changes to the overall class structure in vanagon, most of the work has been done in the class functions to incorporate WiX. Following along the Vanagon::Driver class function “run” :
+
+1. Install_build_dependancies remains unchanged. The underlying changes required to get dependencies for windows lies in the actual project, not vanagon itself.
```

Comment:
Should you put in links to the code examples you are referring to here and generally in this list?